src/agent/src/rpc.rs

```diff
@@ -1906,6 +1917,91 @@ mod tests {
         assert!(result.is_err(), "expected add arp neighbors to fail");
     }
 
+    #[tokio::test]
+    async fn test_get_memory_info() {
+        #[derive(Debug)]
+        struct TestData<'a> {
+            // if None is provided, no files will be generated, else the data in the Option will populate the file
+            block_size_data: Option<&'a str>,
+            hotplug_probe_data: Option<&'a str>,
+
+            result: (u64, bool),
+            error_contains: &'a str,
+        }
+
+        let tests = &[
+            TestData {
+                block_size_data: Some("10000000"),
+                hotplug_probe_data: Some("hotplug data"),
+                result: (268435456, true),
+                error_contains: "",
+            },
+            TestData {
+                block_size_data: Some("100"),
+                hotplug_probe_data: None,
+                result: (256, false),
+                error_contains: "",
+            },
+            TestData {
+                block_size_data: None,
+                hotplug_probe_data: None,
+                result: (0, false),
+                error_contains: "",
+            },
+            TestData {
+                block_size_data: Some(""),
```

Comment:
Could you also add a `TestData` for non numeric data? Something like `Some("foo bar")` to ensure the error path is exercised in that scenario? Yes, it should be "impossible" for scenarios like this to happen, but it never hurts to be paranoid and these sorts of tests are quick to add once you're using the `TestData` approach ;)